### DISCUSSION TITLE
Ability to retry a timed out request

### DISCUSSION THREAD

**Discussion by CameronLonsdale** [projectdiscovery/nuclei#3387] [Q&A] [answered]
### Please describe your feature request:
I'd like for the ability to retry a request which exceeds the timeout value.

Currently I don't believe its possible as retryablehttp library only retries on "retryable" errors https://github.com/projectdiscovery/retryablehttp-go/blob/f406add578abab72be86f1278084717c356e994e/retry.go#L51 which doesn't include the case where the request timed out.

### Describe the use case of this feature:
Sometimes hosts will just timeout, but if you wait and try again the request will succeed. It would be nice to have the option to configure this in nuclei

**Comment by CameronLonsdale** [comment]
You can work around this by just running nuclei multiple times and aggregating the results

**Comment by tarunKoyalwar** [comment, marked as answer]
@CameronLonsdale the desired outcome you mentioned can be achieved by just increasing timeout . 
```console
   -timeout int                        time to wait in seconds before timeout (default 10)
```
the behaviour you mentioned happens if server is slow and retry is not a reliable option for that . it would be best if you just increase the timeout
